File: src/main/node/com/example/Err.ts
/** Error result with a message. */
export default class Err<T> implements Result<T> {
    private readonly message: string;

    Err(message: string): void {
        // TODO
    }

    @Override
    isOk(): boolean {
        return /* TODO */;
    }

    @Override
    value(): Option<any> {
        return /* TODO */;
    }

    @Override
    error(): Option<string> {
        return /* TODO */;
    }
}

File: src/main/java/com/example/Err.java
package com.example;

/** Error result with a message. */
public final class Err<T> implements Result<T> {
    private final String message;

    Err(String message) {
        this.message = message;
    }

    @Override
    public boolean isOk() {
        return false;
    }

    @Override
    public Option<T> value() {
        return Option.none();
    }

    @Override
    public Option<String> error() {
        return Option.some(message);
    }
}
